### PR TITLE
flows: flow cache expiration and benchmarks

### LINKED ISSUE #20: Add benchmark for wrappers and go-dpi layer

### LINKED ISSUE #30: Clean flows older than x minutes automatically from flow tracker

### LINKED ISSUE #41: Some tests failing when wrappers are disabled

### PR DESCRIPTION
**Pull request by nikofil**
* Adds the capability to discard flows after some duration during which
  they have been inactive. (closes #30)

* Create benchmark tests for modules. (closes #20)

* Disable some wrapper tests when the wrappers are disabled. (closes #41)

* Add Go Report Card and fix minor code style issues.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,61 +13,62 @@ It is being developed in the context of the Google Summer of Code 2017 program,
...
 
+For documentation, please check out the [godoc reference](https://godoc.org/github.com/mushorg/go-dpi).
+
 ## Example usage
 
 The library and the modules APIs aim to be very simple and straightforward to use. The library relies on the [gopacket](https://godoc.org/github.com/google/gopacket) library and its Packet structure. Once you have a Packet in your hands, it's very easy to classify it with the library.
-First you need a flow that contains the packet. There is a helper function for constructing a flow from a single packet. Simply call:
+First of all you need to initialize the library. You can do that with simply:
+```go
+godpi.Initialize()
```
Comment (glaslos) [inline README.md]: Maybe add a short explanation what this does?

File: README.md [hunk 2]
```diff
@@ -12,61 +13,62 @@ It is being developed in the context of the Google Summer of Code 2017 program,
 
 Please read the project's [Wiki page](https://github.com/mushorg/go-dpi/wiki) for more information.
 
+For documentation, please check out the [godoc reference](https://godoc.org/github.com/mushorg/go-dpi).
+
 ## Example usage
 
 The library and the modules APIs aim to be very simple and straightforward to use. The library relies on the [gopacket](https://godoc.org/github.com/google/gopacket) library and its Packet structure. Once you have a Packet in your hands, it's very easy to classify it with the library.
-First you need a flow that contains the packet. There is a helper function for constructing a flow from a single packet. Simply call:
+First of all you need to initialize the library. You can do that with simply:
```
Comment (glaslos) [inline README.md]: You use a lot of 'simply', 'easily' 😄 try to keep it a bit more neutral.

File: README.md [hunk 3]
```diff
@@ -12,61 +13,62 @@ It is being developed in the context of the Google Summer of Code 2017 program,
...
+```go
+godpi.Initialize()
+```
+
+Then, you need a flow that contains the packet. You can easily get the flow a packet belongs to. Simply call:
 
 ```go
-flow := godpi.CreateFlowFromPacket(&packet)
+flow, isNew := godpi.GetPacketFlow(&packet)
```
Comment (glaslos) [inline README.md]: Either explain what `isNew` is, or use `_`

File: types/module.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+		filePath := path.Join(dumpsDir, fInfo.Name())
+		dumpPackets, err := utils.ReadDumpFile(filePath)
+		if err != nil {
+			return err
+		}
+		for p := range dumpPackets {
+			flow, _ := GetFlowForPacket(&p)
+			if flow.DetectedProtocol == Unknown {
+				module.ClassifyFlow(flow)
+			}
```
Comment (glaslos) [inline types/module.go]: I was actually thinking about having the benchmark measure calls to `ClassifyFlow`

Comment (nikofil) [inline types/module.go]: You mean, even if the flow is already classified? Or the generic `ClassifyFlow` than runs both classifiers and wrappers?

File: README.md (hunk 2 above)
Comment (nikofil) [inline README.md]: Alright :) I'll push these changes to README.md to my other PR, as that's where that commit belongs, and rebase this PR once the other one is merged.

Comment (nikofil) [inline README.md]: Actually I just rebased it anyway, to have the changes here too.

**Comment by glaslos** [comment]
Needs a rebase

File: types/module.go (hunk 4 above)
Comment (glaslos) [inline types/module.go]: Sorry, meant the generic `ClassifyFlow` as a beginning and then for all the different classifiers individually.
```go
func BenchmarkClassifyFlow(b *testing.B) {
        for n := 0; n < b.N; n++ {
                godpi.ClassifyFlow(flow)
        }
}
```

**Pull request merged**